libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -625,18 +625,20 @@ void AP_GPS::update_instance(uint8_t instance)
     // has expired, re-initialise the GPS. This will cause GPS
     // detection to run again
     if (!result) {
-        if (tnow - timing[instance].last_message_time_ms > GPS_TIMEOUT_MS) {
-            // free the driver before we run the next detection, so we
-            // don't end up with two allocated at any time
-            delete drivers[instance];
-            drivers[instance] = nullptr;
-            memset(&state[instance], 0, sizeof(state[instance]));
-            state[instance].instance = instance;
-            state[instance].status = NO_GPS;
-            state[instance].hdop = GPS_UNKNOWN_DOP;
-            state[instance].vdop = GPS_UNKNOWN_DOP;
-            timing[instance].last_message_time_ms = tnow;
-            timing[instance].delta_time_ms = GPS_TIMEOUT_MS;
+        // do not try to detect again if type is not auto
+        if (_type[instance] == GPS_TYPE_AUTO) {
+            if (tnow - timing[instance].last_message_time_ms > GPS_TIMEOUT_MS) {
+                // free the driver before we run the next detection, so we
+                // don't end up with two allocated at any time
+                delete drivers[instance];
+                drivers[instance] = nullptr;
+                memset(&state[instance], 0, sizeof(state[instance]));
+                state[instance].instance = instance;
+                state[instance].status = NO_GPS;
+                state[instance].hdop = GPS_UNKNOWN_DOP;
+                state[instance].vdop = GPS_UNKNOWN_DOP;
+                timing[instance].delta_time_ms = GPS_TIMEOUT_MS;
```

Comment:
Was removing this line intentional?
`timing[instance].last_message_time_ms = tnow;`



if you change the line to be a single if then the whitespace won't change and you would have seen that change
`if (_type[instance] == GPS_TYPE_AUTO && (tnow - timing[instance].last_message_time_ms > GPS_TIMEOUT_MS))`

Reply:
`tnow` here is redundant, it'll be set by `detect_instance` in the next `update_instance`.

Reply:
So I think it should probably be set, as it's the debounce time that keeps you from attempting to free/memset the GPS driver/state every 20ms.